### PR TITLE
Add grape/swagger defaultValue support

### PR DESCRIPTION
**Pull request by jv-dan**
See issue #215

### CODE REVIEW

**Comment by dblock** [comment]
RuboCop! please fix the build to start.

**Comment by dblock** [comment]
This needs a CHANGELOG entry and a README update, please.

**Comment by jv-dan** [comment]
rubuchangeme updated.

**Comment by dblock** [comment]
Would you please squash these? I'll do a thorough review soon.

File: README.md [hunk 1]
```diff
@@ -229,6 +229,30 @@ The name should not contain whitespaces or any other special characters due to f
...
 
+## Additional documentation
+
+## Setting a Swagger defaultValue
+
+Grape allows for an additional documentation hash to be passed to a parameter.
+
+    params do
+      requires :id, type: Integer, desc: 'Coffee ID'
+      requires :temperature, type: Integer, desc: 'Temperature of the coffee in celcius', documentation: {example: 72}
```
Comment (dblock) [inline README.md]: Space out `{ example: 72 }`.

File: spec/grape-swagger_helper_spec.rb [hunk 2]
```diff
@@ -24,6 +24,19 @@ class HelperTestAPI < Grape::API
       ]
     end
 
+    it 'parses params as query strings for a GET with a default_value' do
```
Comment (dblock) [inline spec/grape-swagger_helper_spec.rb]: with an example, right? not default_value

File: spec/grape-swagger_helper_spec.rb [hunk 3]
```diff
@@ -24,6 +24,19 @@ class HelperTestAPI < Grape::API
...
+      params = {
+        name: { type: 'String', desc: 'A name', required: true, default: 'default api value', documentation: {example: 'default swagger value'} },
```
Comment (dblock) [inline spec/grape-swagger_helper_spec.rb]: Indent hash please and remove it from Rubocop's list of acceptable violations.

**Comment by dblock** [comment]
Would you mind squashing these and rebasing? There were a bunch of changes upstream. Add a comment so I can merge this after, please. Sorry for the hassle.

**Comment by jv-dan** [comment]
Squashed/rebased. no problem at all.

File: CHANGELOG.md [hunk 4]
```diff
@@ -11,6 +11,7 @@
 
 #### Features
 
+* [#215](https://github.com/tim-vandecasteele/grape-swagger/pull/217): Support swagger `defaultValue` without the need to set a Grape `default` - [@jv-dan](https://github.com/jv-dan).
```
Comment (dblock) [inline CHANGELOG.md]: This was already released, move this up to Next please.

File: README.md [hunk 5]
```diff
@@ -229,6 +229,30 @@ The name should not contain whitespaces or any other special characters due to f
...
+
+    params do
+      requires :id, type: Integer, desc: 'Coffee ID'
+      requires :temperature, type: Integer, desc: 'Temperature of the coffee in celcius', documentation: { example: 72 }
+    end
+
+The example parameter will populate the Swagger UI with the example value, and can be used for optional or required parameters.
+
+Grape uses the option `default` to set a default value for optional parameters. This is different in that Grape will set your parameter
+ to the provided default if the parameter is omitted, whereas the example value above will only set the value in the UI itself. This will
```
Comment (dblock) [inline README.md]: Super minor. Just make this one line while you're at it, don't break at 80 characters.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,6 +1,6 @@
 ### 0.10.2 (Next)
 
-* Your contribution here.
```
Comment (dblock) [inline CHANGELOG.md]: Put back that one, below! Thanks.

**Comment by jv-dan** [comment]
Should be all set once Travis does his thing

**Comment by dblock** [comment]
Merged, thank you.

**Pull request merged**